### PR TITLE
[LoongArch] Enable alias analysis by default

### PR DESCRIPTION
**Pull request by heiher**
Enable use of alias analysis during code generation.

### CODE REVIEW

File: llvm/test/CodeGen/LoongArch/merge-load-store.ll
Comment (SixWeining) [inline llvm/test/CodeGen/LoongArch/merge-load-store.ll:1]: > define void @merge_load_store(ptr noalias nocapture noundef readonly align 1 dereferenceable(8) %src, ptr noalias nocapture noundef writeonly align 1 dereferenceable(8) %dst) unnamed_addr {

Are all these attributes beside `noalias` necessary for this test?

Comment (heiher) [reply llvm/test/CodeGen/LoongArch/merge-load-store.ll:1]: No. I'll reland the pre-commit test to remove these attributes.

**Review by SixWeining — APPROVED** [review]
LGTM

**Review by wangleiat — APPROVED** [review]
Thanks

**Pull request merged**